### PR TITLE
move the assertions outside of the goroutine

### PR DESCRIPTION
**Pull request by mountkin**
I mistakenly added some assertions in a gogoutine in integration test when trying to fix a panic several days before :sob:  This PR tries to fix it.

BTW: can we completely remove this test case? It's just here to test a fixed panic.

Signed-off-by: Shijiang Wei mountkin@gmail.com

### CODE REVIEW

**Comment by calavera** [comment]
Thanks a lot for doing this.

LGTM

File: integration-cli/docker_api_exec_resize_test.go [hunk 1]
```diff
@@ -28,53 +29,75 @@ func (s *DockerSuite) TestExecResizeImmediatelyAfterExecStart(c *check.C) {
...
 			}
 		}()
 	}
 
-	for i := 0; i < n; i++ {
-		<-ch
+	wg.Wait()
+	close(ch)
+
+	for err := range ch {
```
Comment (LK4D4) [inline integration-cli/docker_api_exec_resize_test.go]: You interested only in one error anyway, so it can be just:
```
c.Assert(<-ch, check.IsNil)
```

Comment (mountkin) [inline integration-cli/docker_api_exec_resize_test.go]: Thanks @LK4D4 
Receiving from a closed channel seems a bit odd (although it's ok when there are data to receive). I updated the PR to removed the `close` and check the error in a `select` statement.

**Comment by cpuguy83** [comment]
Thanks, LGTM!

**Pull request merged**